prusti-encoder/src/encoders/type/kinds/immref.rs

```diff
@@ -29,12 +31,19 @@ pub(crate) fn domain<'vir>(
     let value_ident = builder.function("value", &[builder.self_type()], inner_type);
     let cons_ident = builder.function("cons", &[prim_type, inner_type], builder.self_type());
 
+    let generic_enc = deps.require_ref::<GenericEnc>(())?;
+    let ty_type_func = deps.require_ref::<TyConstructorEnc>(task_key)?;
     builder.axiom("deref", vir::expr! {
         forall r: [prim_type], value: [inner_type] :: {[cons_ident](r, value)} ([deref_ident]([cons_ident](r, value))) == (r)
     });
     builder.axiom("value", vir::expr! {
         forall r: [prim_type], value: [inner_type] :: {[cons_ident](r, value)} ([value_ident]([cons_ident](r, value))) == (value)
     });
+    builder.axiom("ref_immutable_cons_typeof", vir::expr! {
```

Comment:
```suggestion
    builder.axiom("typeof", vir::expr! {
```
(the names of axioms built using the builder are all prefixed with the type name already)